core/src/test/scala/org/apache/spark/deploy/history/HistoryServerSuite.scala

```diff
@@ -147,6 +152,58 @@ class HistoryServerSuite extends FunSuite with BeforeAndAfter with Matchers with
     }
   }
 
+  test("download all logs for app with multiple attempts") {
+    doDownloadTest(None)
+  }
+
+  test("download one log for app with multiple attempts") {
+    (1 to 2).foreach { attemptId => doDownloadTest(Some(attemptId)) }
+  }
+
+  // Test that the files are downloaded correctly, and validate them.
+  def doDownloadTest(attemptId: Option[Int]): Unit = {
+
+    val url = attemptId match {
+      case Some(id) =>
+        new URL(s"${generateURL("applications/local-1430917381535")}/$id/logs")
+      case None =>
+        new URL(s"${generateURL("applications/local-1430917381535")}/logs")
+    }
+
+    val (code, inputStream, error) = HistoryServerSuite.connectAndGetInputStream(url)
+    code should be (HttpServletResponse.SC_OK)
+    inputStream should not be None
+    error should be (None)
+
+    def validateFile(fileName: String, tempDir: File): Unit = {
+      val inStream = new FileInputStream(new File(logDir, fileName))
+      val outStream = new FileInputStream(new File(tempDir, fileName))
```

Comment:
the names `inStream` & `outStream` are a little misleading, maybe `expStream` and `actualStream`?